sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TelemetryPartCTests.cs

```diff
@@ -0,0 +1,244 @@
+                ActivitySpanId.CreateRandom(),
+                ActivityTraceFlags.None), null);
+
+            List<ActivityLink> links = new List<ActivityLink>();
+            links.Add(activityLink);
+
+            using var activity = activitySource.StartActivity(
+                ActivityName,
+                ActivityKind.Client,
+                parentContext: default,
+                null,
+                links,
+                startTime: DateTime.UtcNow);
+
+            string expectedMSlinks = GetExpectedMSlinks(links);
+            var telemetryPartBRequestData = TelemetryPartB.GetRequestData(activity);
+
+            Assert.True(telemetryPartBRequestData.Properties.TryGetValue(msLinks, out var actualMSlinks));
+            Assert.Equal(expectedMSlinks, actualMSlinks);
+        }
+
+        [Fact]
+        public void RemoteDependencyDataPropertiesDoesNotContainMSLinksWhenActivityHasNoLinks()
+        {
+            using ActivitySource activitySource = new ActivitySource(ActivitySourceName);
+            using var activity = activitySource.StartActivity(
+                ActivityName,
+                ActivityKind.Client,
+                parentContext: default,
+                startTime: DateTime.UtcNow);
+
+            var telemetryPartBRemoteDependencyData = TelemetryPartB.GetRemoteDependencyData(activity);
+            Assert.False(telemetryPartBRemoteDependencyData.Properties.TryGetValue(msLinks, out var mslinks));
+        }
+
+        [Fact]
+        public void RemoteDependencyDataPropertiesContainsMSLinksWhenActivityHasLinks()
+        {
+            using ActivitySource activitySource = new ActivitySource(ActivitySourceName);
+            ActivityLink activityLink = new ActivityLink(new ActivityContext(
+                ActivityTraceId.CreateRandom(),
+                ActivitySpanId.CreateRandom(),
+                ActivityTraceFlags.None), null);
+
+            List<ActivityLink> links = new List<ActivityLink>();
+            links.Add(activityLink);
+
+            using var activity = activitySource.StartActivity(
+                ActivityName,
+                ActivityKind.Client,
+                parentContext: default,
+                null,
+                links,
+                startTime: DateTime.UtcNow);
+
+            string expectedMSlinks = GetExpectedMSlinks(links);
+            var telemetryPartBRemoteDependencyData = TelemetryPartB.GetRemoteDependencyData(activity);
+
+            Assert.True(telemetryPartBRemoteDependencyData.Properties.TryGetValue(msLinks, out var actualMSlinks));
+            Assert.Equal(expectedMSlinks, actualMSlinks);
+        }
+
+        [Fact]
+        public void LinksAreTruncatedWhenCannotFitInMaxLength()
+        {
+            using ActivitySource activitySource = new ActivitySource(ActivitySourceName);
+            List<ActivityLink> links = new List<ActivityLink>();
+
+            int numberOfLinks = 150; //arbitrary number > 100
+            int maxLinksAllowed = 100;
```

Comment:
don't define them define. refer from a common place.

Reply:
maybe make it internal from the exporter, and make the exporter InternalsVisibleTo the tests.

Reply:
Yes - I am going to find a common place for all such constraints when I add the complete truncation logic. Will move it then